### PR TITLE
SELinux support for ceph

### PR DESCRIPTION
**Pull request by branto1**
The patches are getting there, we might still need to fine-tune the SELinux policy this but you can already review the build system and .spec file changes.

### CODE REVIEW

**Comment by liewegas** [comment]
Let's get this through qa so we can hopefully merge

File: ceph.spec.in [hunk 1]
```diff
@@ -61,6 +78,11 @@ Requires: systemd
 %if 0%{with cephfs_java}
 BuildRequires: sharutils
 %endif
+%if 0%{with selinux}
+BuildRequires:	checkpolicy
+BuildRequires:	selinux-policy-devel
+BuildRequires:	/usr/share/selinux/devel/policyhelp
```
Comment (tchaikov) [inline ceph.spec.in]: - this file belongs to ”selinux-policy-doc“ on my fedora22. but ”selinux-policy-doc“ is not listed in the BuildRequires.
- and this file is not needed anymore to figured out the `_selinux_policy_version` on fedora 20+.

Comment (branto1) [inline ceph.spec.in]: @tchaikov: The last time I checked, the.spec file built fine on f22 (it was a few commits back but it already contained this chunk). Either way, I scheduled a koji build on top of this branch to confirm:

http://koji.fedoraproject.org/koji/taskinfo?taskID=10598873

Comment (tchaikov) [inline ceph.spec.in]: thanks! i am testing it on our centos 6.5 gitbuilder.

Comment (branto1) [inline ceph.spec.in]: np, btw: the rpms were built successfully.

Comment (tchaikov) [inline ceph.spec.in]: @branto1 i have following errror
```
make -j1 -f /usr/share/selinux/devel/Makefile ceph.pp
cat: /selinux/mls: No such file or directory
```

when `make`ing in the `selinux` directory, after installing `selinux-policy-doc`. shall we add `selinux-policy-mls` to the build-deps as well?

Comment (branto1) [inline ceph.spec.in]: I do not need to have mls enabled for this. I was trying to run this on a range of a more recent fedoras (21+) and rhel/centos 7 only.

Anyway, did the installation of that package fixed your issue? If yes then we could probably add it to the dep list, at least for rhel6-.

**Comment by branto1** [comment]
FWIW: We are currently discussing the future of the SELinux policy and it will probably go to the upstream selinux-policy package/project so do not merge these changes upstream just yet, please.

Nevertheless, this is probably the way it will be back-ported to some of the next rhceph products so feel free to review especially the spec file changes.

File: ceph.spec.in (hunk 1 above)
Comment (tchaikov) [inline ceph.spec.in]: @branto1 i checked the `/usr/share/selinux/devel/Makefile` on centos6.6 and f22, seems `mls` is chosen by default. will update this pr once http://gitbuilder.sepia.ceph.com/gitbuilder-ceph-rpm-centos6-5-amd64-basic/#origin/wip-kefu-testing is finished.

**Comment by tchaikov** [comment]
@branto1 ack. will remove it from my qa run.

**Comment by branto1** [comment]
tchaikov: Actually, the qa run would be very much appreciated at this moment (although I'm not sure what exactly does the qa run covers).

**Comment by tchaikov** [comment]
@branto1 the qa run will at least ensure that 

- the tree builds on gitbuilders.
- the rpm/deb on gitbuilder builds
- the "make check" passes on gitbuilders
- and the qa suite passes. in my case, i am running rados qa suite.

to this pr, probably what matters most is 
- the tree builds on all gitbuilders.
- the rpm/deb on all gitbuilder builds

File: ceph.spec.in (hunk 1 above)
Comment (tchaikov) [inline ceph.spec.in]: @branto1 i still this error. `selinux-policy-mls` does not help.

see http://paste2.org/nBc4bxY7

Comment (branto1) [inline ceph.spec.in]: The force rebuild won't help, here. The issue is that the policy packages in centos 6.5 do not have a certain macro defined. We can probably disable the (default) SELinux support in centos 6 for now. If there is a demand for SELinux on rhel 6 then we should work on getting the appropriate SELinux bits, there.

**Comment by branto1** [comment]
OK, after discussion with selinux-policy maintainers, etc, we decided to maintain the policy in ceph git repo. Hence, I removed the DNM tag from this PR.

**Comment by mbroz** [comment]
@tchaikov than policy build error for CentOS6 - acording to selinux maintainer, fsadm_manage_pid macro will be backported later (CentOS 6.8?) but I think we can ignore RHEL/CentOS6 for now

**Comment by liewegas** [comment]
On Tue, 4 Aug 2015, Milan Broz wrote:
> @tchaikov than policy build error for CentOS6 - acording to selinux maintainer, fsadm_manage_pid macro will be backported later (CentOS 6.8?) but
> I think we can ignore RHEL/CentOS6 for now

Agree.

We're about to break el6 with c++11 anyway.

**Comment by branto1** [comment]
btw: I rebased this branch to resolve the merge conflict (systemd patches added new directory, that is all).

File: ceph.spec.in (hunk 1 above)
Comment (tchaikov) [inline ceph.spec.in]: ah, i see.

**Comment by tchaikov** [comment]
> @tchaikov than policy build error for CentOS6 - acording to selinux maintainer, `fsadm_manage_pid` macro will be backported later (CentOS 6.8?) but I think we can ignore RHEL/CentOS6 for now

thanks. got it.

**Comment by branto1** [comment]
I (with help of Yuri) reviewed the smoke test results and I believe that we are good to merge this. There were some SELinux denials but those were caused by test case problem, not our policy.

**Pull request merged**